docs/py3_todo.md

```diff
@@ -0,0 +1,29 @@
+#Python 3 TODO
+List of known issues, bugs and todos for stash py3 (and py2) compatibility.
+
+#General
+- ctype threads in py3 bugged (crash on cc and kill)
+- more commands need to be ported
+- sys.argv needs to be bytestr in py2 and unistt in py3. There is only a quick fix in place (somewhere in `shruntime.py`), which should be replaced as it only works for common usage situations
+- unittests for py3
+- i/o seems to switch between jobs from time to time
+- 'ValueError: I/O operation on closed file' prevents some pipes from working
+
+
+#`crypt.py`
+- in py3, the key is shown as `b'<key>'` instead of `<key>`. Also test solution with py2
+
+#`curl.py`
+- more testing
+
+#`easy_config.py`
+- while syntax is py3 compatible, there are some errors
+- untested
+
+#`pip.py`
+- many bugs
+- installation using `setupy.py` fails most of the time, always falls back to package detection
+- maybe `pip3` for py3 instead of a single `pip` command.
```

Comment:
If we _need_ two different pips, then we should make three instead (like homebrew does)?
* __pip2__ always installs into site packages for Python 2
* __pip3__ always installs into site packages for Python 3
* __pip__ looks to see what Python is currently running and then uses __pip2__ or __pip3__ accordingly.

Reply:
@cclauss Good idea. I dont think we actually *need* two different pips. It seems like we *could* just use the same one. The `pip3` proposal was only meant for consistency with the `python` and `python3` commands.

Reply:
Good if we only have one pip command.  The Python command could operate the same way...
* __python2__ works on Python __2__ but fails with an error message on Python __3__
* __python3__ works on Python __3__ but fails with an error message on Python __2__
* __python__ looks to see what Python is currently running and then uses python2 or python3 accordingly.

Much like homebrew https://docs.brew.sh/Homebrew-and-Python

Reply:
By the way, this is specified by [PEP 394](https://www.python.org/dev/peps/pep-0394/); it's not specific to Homebrew.